R/utils.R

```diff
@@ -1,3 +1,20 @@
+recodeOverlayValues <- function(overlayData, desiredOverlayValues, variables) {
+  if (is.null(desiredOverlayValues)) return(overlayData)
+  dataType <- veupathUtils::findDataTypesFromPlotRef(variables, 'overlay')
+  
+  if (dataType %in% c('NUMBER', 'INTEGER', 'DATE')) {
+    # figure this out when i get to the continuous overlay PR, just leaving a skeleton for now
+    # tbh desiredOverlayValues will probably stop being a simple character vector then as well. but one thing at a time.
+    warning("Binned continuous overlays are not supported yet.")
+  } else {
+    if (unique(overlayData) == desiredOverlayValues) return(overlayData)
```

Comment:
Does the order of `unique(overlayData)` have to be the same as `desiredOverlayValues`?

If so, something like `sort(unique(overlayData)) == sort(desiredOverlayValues)` ??

But note my lack of R prowess!

Reply:
I should be using all() and %in%